core/model/modx/mysql/modcontext.class.php

```diff
@@ -12,33 +12,88 @@ class modContext_mysql extends modContext {
     public static function getResourceCacheMapStmt(&$context) {
         $stmt = false;
         if ($context instanceof modContext) {
+            $time=microtime(true);
+            $cache_alias_map = $context->getOption('cache_alias_map');
+            $use_context_resource_table = $context->getOption('use_context_resource_table',null,1);
+
             $tblResource= $context->xpdo->getTableName('modResource');
             $tblContextResource= $context->xpdo->getTableName('modContextResource');
+
             $resourceFields= array('id','parent','uri');
+
+            // we do not need to select uri if cache_alias_map is set to false
+            if ($cache_alias_map==0) $resourceFields = array('id','parent');
```

Comment:
Code style should be more like

`if ($cache_alias_map == 0) {
    $resourceFields = ..
}`